primitives/src/test_data/sector_timeline.rs

```diff
@@ -0,0 +1,230 @@
+        if publish_storage_deals >= pre_commit_sectors {
+            return Err(format!(
+                "publish_storage_deals ({:?}) must be less than pre_commit_sectors ({:?})",
+                publish_storage_deals, pre_commit_sectors
+            ));
+        }
+        let calculated_period = challenge_window * period_deadlines;
+        if calculated_period.0 != proving_period.0 {
+            return Err(format!(
+                "calculated period ({:?}) must be equal to proving period ({:?})",
+                calculated_period, proving_period
+            ));
+        }
+        let t = Self {
+            storage_provider_account_id,
+            register_storage_provider,
+            publish_storage_deals,
+            pre_commit_sectors,
+            deals,
+
+            period_deadlines,
+            proving_period,
+            challenge_window,
+            pre_commit_challenge_delay,
+            challenge_lookback,
+        };
+
+        if t.proving_period_start_initial() >= t.seal_randomness_height() {
+            return Err(format!(
+                "proving period start ({:?}) must be less than seal randomness height ({:?}), try moving the pre_commit_sectors block forward",
+                t.proving_period_start_initial().0,
+                t.seal_randomness_height().0
+            ));
+        }
+        if t.deadline_start() < t.proving_period_start_initial() {
+            return Err(format!(
+                "deadline start ({:?}) must be greater than or equal to the initial proving period start ({:?}), try moving the deal start block forward",
+                t.deadline_start().0,
+                t.proving_period_start_initial().0
+            ));
+        }
+
+        for deal in &t.deals {
+            if deal.start() < t.prove_commit_sectors() {
+                return Err(format!(
+                    "deal start ({:?}) must be greater than or equal to the initial proving period start ({:?}), try moving the deal start block forward",
+                    deal.start().0,
+                    t.prove_commit_sectors().0
+                ));
+            }
+        }
+
+        if t.deadline_challenge_block() < t.prove_commit_sectors() {
+            return Err(format!(
+                "deadline start ({:?}) must be greater than or equal to the initial proving period start ({:?}), try moving the deal start block forward",
+                t.deadline_start().0,
+                t.proving_period_start_initial().0
+            ));
+        }
+
+        Ok(t)
+    }
+
+    pub fn storage_provider_account_id(&self) -> &AccountId {
+        &self.storage_provider_account_id
+    }
+
+    pub fn register_storage_provider(&self) -> Absolute<BlockNumber> {
+        self.register_storage_provider
+    }
+
+    pub fn publish_storage_deals(&self) -> Absolute<BlockNumber> {
+        self.publish_storage_deals
+    }
+
+    pub fn proving_period(&self) -> Relative<BlockNumber> {
+        self.proving_period
+    }
```

Comment:
Do we actually need these getters? Since this is test data it should be ok to just expose them (?)

Reply:
Fair enough